src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java

```diff
@@ -877,10 +878,13 @@ private static void validatePropertySectionPropertyTokens(String fileName, Strin
                         + "' should have the basic token description");
 
         final String acceptableTokenText = columns.get(2).getTextContent().trim();
-        assertEquals("subset of tokens "
-                        + CheckUtil.getTokenText(check.getAcceptableTokens(),
-                check.getRequiredTokens()),
-                acceptableTokenText
+        String expectedAcceptableTokenText = "subset of tokens "
+                + CheckUtil.getTokenText(check.getAcceptableTokens(),
+                check.getRequiredTokens());
+        if ("DescendantToken".equals(sectionName)) {
```

Comment:
```
tokens  = CheckUtil.getTokenText(check.getAcceptableTokens(),
                check.getRequiredTokens())
if (tokens.size() == TOKEN_TYPE_TOTAL_COUNT) {
 expectedAcceptableTokenText = "all tokens allTokenTypes";
} else {
 expectedAcceptableTokenText = "subset of tokens "
                + tokens;
}
```

Reply:
But `DescendantTokenCheck` will satisfy that `tokens.size() != TokenUtil.getAllTokenIds().length`, and the test will fail. tokens.size() is zero.

Reply:
`check.getRequiredTokens()` is empty for Check. getAcceptableTokens are all tokens. So `tokens` will be all tokens.

Am I miss something.

Reply:
fixed.
One logic miss hardcode ... new problem was found and fixed.